daemon/info.go

```diff
@@ -119,8 +124,8 @@ func (daemon *Daemon) SystemVersion() types.Version {
 
 	v.Platform.Name = dockerversion.PlatformName
 
-	daemon.fillPlatformVersion(&v)
-	return v
+	daemon.fillPlatformVersion(ctx, &v)
+	return v, ctx.Err()
```

Comment:
Could use some eyes on these; perhaps a silly question; should these context errors be handled by the calling site (checking if there was a context-error), instead of them being returned as error here?

Reply:
Unless we are actually supporting cancellation within the function, I don't think it makes sense to return a context error.

Reply:
@cpuguy83 cancellation _is_ supported: the expensive operations within `daemon.fillPlatformVersion` are aborted/skipped if the context is done. As that leaves `v` partially or fully unfilled, returning a non-nil error when the context is done is necessary to signal to the caller that the returned `v` is not valid.

Reply:
Then that function should return the error.
The current way this is done is racey because the call could have succeeded but then got cancelled after but before we actually return.

Reply:
So what if it's racey? It doesn't matter; all operations which take contexts race to complete vs context cancelation. Consider the case where the race resolves the other way, with `daemon.SystemVersion()` returning before the context is canceled. The context is still canceled; the return value will be discarded regardless because the client has dropped (for example) and there's nowhere to write the response.

Modifying `daemon.fillPlatformVersion` to return the error won't eliminate the race. And that's easier said than done, too: the error handling in there is currently "log a warning and continue on regardless." There is no way to differentiate between `(*exec.Cmd).Output()` returning an error because the the context was canceled and the process being SIGKILLed for any other reason without racing context-canceled-before-return. The error handling would need to change in observable ways for the non-canceled case.